aesara/tensor/subtensor_opt.py

```diff
@@ -1666,3 +1668,98 @@ def local_subtensor_SpecifyShape_lift(fgraph, node):
         new_shape_arg = as_tensor(shape_arg[indices], dtype=np.int64, ndim=1)
 
     return [specify_shape(new_obj_arg, new_shape_arg)]
+
+
+def equal_idxs(idx1, idx2):
+    if isinstance(idx1, slice) and isinstance(idx2, slice):
+        return (
+            equal_computations([idx1.start], [idx2.start])
+            and equal_computations([idx1.stop], [idx2.stop])
+            and equal_computations([idx1.step], [idx2.step])
```

Comment:
If the sub-graphs that evaluate to `True` under `equal_computations` can be merged by `MergeOptimizer`, then use of `equal_computations` can be wasteful (remember, it may need to walk the entire graph).  In such cases, it might be better to use identity comparisons and wait until the sub-graphs have been merged.

Reply:
That seems to work just fine! Should we still leave the rewrite as `specialize` only?

Reply:
> That seems to work just fine! Should we still leave the rewrite as `specialize` only?

It's not a question of whether `equal_computations` works, but one of whether or not it also works _without_ `equal_computations` and its associated costs.

Reply:
All the tests I had written worked out of the box...

Reply:
Without it?

Reply:
My question is whether this makes the rewrite light enough to justify being a canonicalization. I assumed that was the concern before for making it only a specialization

Reply:
> Without it?

Yes. I am no longer using any `equal_computations` in the new rewrite

Reply:
This is a tough one, because it's a justifiable canonicalization (given our current standard, at least), and it's not completely clear which approach (i.e. `equal_computations`/no `equal_computations) is better on average&mdash;at least in terms of cost.

It's always possible that using `equal_computations` could avoid a few intermediate steps (e.g. the merge steps needed to make the identity-based comparisons `True`, and possibly more), but that has to be offset by the cost of using `equal_computations` in all other cases.

Reply:
As a default, I think it's better to work within the rewrite pipeline, and `equal_computations` has the potential to "override" the basic rewrite pass logic.  For example, given a fixed set of deterministically ordered rewrite passes, another rewrite might be expected to act upon a graph that would've been rewritten under the use of `equal_computations`.  In other words, use of `equal_computations` has the potential to preempt the set of rewrite passes that would've otherwise been applied, because it effectively performs a merge out-of-sequence.